sycl/include/sycl/ext/intel/experimental/esimd/detail/memory_intrin.hpp

```diff
@@ -646,6 +660,14 @@ __esimd_gather_masked_scaled2(SurfIndAliasTy surf_ind, uint32_t global_offset,
 {
   static_assert(Scale == 0);
 
+  // ActualTy - Reverse of 'PromoT' in memory.hpp
+  using ActualTy = std::conditional_t<
+      __SEIEED::ElemsPerAddrEncoding<sizeof(Ty)>() == TySizeLog2, Ty,
```

Comment:
here and in other places
please use ElemsPerAddrDecoding instead of directly using TySizeLog2

Reply:
Done.